doc/lua_api.txt

```diff
@@ -617,6 +640,64 @@ Result is more like what you'd expect if you put a color on top of another
 color, meaning white surfaces get a lot of your new color while black parts
 don't change very much.
 
+A Multiply blend can be applied between two textures by using the overlay 
+modifier with a brightness adjustment:
+
+    textureA.png^[contrast:0:-64^[overlay:textureB.png
+
+#### `[screen:<color>`
+
+Apply a Screen blend with the given color. A Screen blend is the inverse of
+a Multiply blend, lightening images instead of darkening them.
+
+`<color>` is specified as a `ColorString`.
+
+A Screen blend can be applied between two textures by using the overlay 
+modifier with a brightness adjustment:
+
+    textureA.png^[contrast:0:64^[overlay:textureB.png
+
+#### `[hsl:<hue>:<saturation>:<lightness>`
+
+Adjust the hue, saturation, and lightness of the texture. Like
+"Hue-Saturation" in GIMP, but with 0 as the mid-point.
+
+`<hue>` should be from -180 to +180
+
+`<saturation>` and `<lightness>` are optional, and both from -100 to +100
+
+#### `[contrast:<contrast>:<brightness>`
```

Comment:
brightness should IMO be an optional argument of `brighten` instead of mushed in here

Reply:
They are not mushed together here arbitrarily. Brightness and contrast are the two knobs of the same operation, they are together here for the same reason they are together in gimp and photoshop.

Changing the slope of an adjustment line/curve changes contrast, and changing its y-intercept changes brightness. Changing them together instead of running two separate adjustment operations reduces rounding/quantization error and prevents any clipping from applying operations in a bad order. 

Usually you want the contrast to change automatically as you change brightness to avoid clipping, so conventional brightness *is* a contrast operation. But at this point I'm not sure what the "correct" UI is for the two knobs (or whether there even is a correct way), my implementation treats your contrast value as how much you wish to rotate the adjustment line, and that rotation is applied on top of any slope given to the adjustment line by the change in brightness (if one is specified).